Buildbarn test now failing

After much effort to get the BuildGrid test to work again in #1582 ... the updated docker images in #1612 now causes the buildbarn remote cache test to fail :(

Running the test `./.github/run-ci.sh  -s buildbarn` gives me the following failure:

```
________________________________________________________________________________ test_remote_autotools_build _________________________________________________________________________________

cli = <buildstream._testing.runcli.CliRemote object at 0x7f5e26d645e0>, datafiles = local('/home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0')

    @pytest.mark.datafiles(DATA_DIR)
    def test_remote_autotools_build(cli, datafiles):
        project = str(datafiles)
        checkout = os.path.join(cli.directory, "checkout")
        element_name = "autotools/amhello.bst"
    
        result = cli.run(project=project, args=["build", element_name])
        result.assert_success()
>       assert element_name in result.get_pushed_elements()
E       AssertionError: assert 'autotools/amhello.bst' in []
E        +  where [] = <bound method Result.get_pushed_elements of <buildstream._testing.runcli.Result object at 0x7f5e26cdcee0>>()
E        +    where <bound method Result.get_pushed_elements of <buildstream._testing.runcli.Result object at 0x7f5e26cdcee0>> = <buildstream._testing.runcli.Result object at 0x7f5e26cdcee0>.get_pushed_elements

tests/remotecache/simple.py:25: AssertionError
------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------
BuildStream exited with code 0 for invocation:
	bst --no-colors --config /home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0/cache/buildstream.conf --directory /home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0 build autotools/amhello.bst
Program stderr was:
[--:--:--][        ][    main:core activity                 ] START   Build
[--:--:--][        ][    main:core activity                 ] START   Loading elements
[00:00:00][        ][    main:core activity                 ] SUCCESS Loading elements

    3 subtasks processed

[--:--:--][        ][    main:core activity                 ] START   Resolving elements
[00:00:00][        ][    main:core activity                 ] SUCCESS Resolving elements

    3 of 3 subtasks processed

[--:--:--][        ][    main:core activity                 ] START   Initializing remote caches
[00:00:00][        ][    main:core activity                 ] SUCCESS Initializing remote caches
[--:--:--][        ][    main:core activity                 ] START   Checking sources
[00:00:00][        ][    main:core activity                 ] SUCCESS Checking sources
[--:--:--][        ][    main:core activity                 ] START   Query cache
[00:00:00][        ][    main:core activity                 ] SUCCESS Query cache
[--:--:--][        ][    main:core activity                 ] START   Preparing work plan
[00:00:00][        ][    main:core activity                 ] SUCCESS Preparing work plan

    3 of 3 subtasks processed


BuildStream Version 1.93.6+253.g56ab9fae2
    Session Start: Saturday, 12-03-2022 at 05:15:36
    Project:       test (/home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0)
    Targets:       autotools/amhello.bst

User Configuration
    Configuration File:      /home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0/cache/buildstream.conf
    Cache Directory:         /home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0/cache
    Log Files:               /home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0/cache/logs
    Source Mirrors:          /home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0/cache/sources
    Build Area:              /home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0/cache/build
    Strict Build Plan:       Yes
    Maximum Fetch Tasks:     10
    Maximum Build Tasks:     4
    Maximum Push Tasks:      4
    Maximum Network Retries: 2

Project: test

    Project Options
        arch:  x86-64
        linux: 1

    Element Plugins
        autotools: core plugin
        stack:     core plugin
        import:    core plugin

    Source Plugins
        tar: core plugin

    Artifact cache servers
      * http://localhost:7981
        push: True type: index instance: None
        loaded from: /home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0/cache/buildstream.conf [line 3 column 4]
      * http://localhost:7982
        push: True type: storage instance: None
        loaded from: /home/testuser/buildstream/.tox/py310-nocover/tmp/test_remote_autotools_build0/cache/buildstream.conf [line 6 column 4]

Pipeline
fetch needed a1e46db9db777559a74aca850f6f58c5745ce36b61ed75a26e8056d4a0b967b1 base/base-alpine.bst 
     waiting 378a7f2d90b64643bb42bc3f3c6252f0304c3d73ce78cf1a9af9a01ec47fc5d8 base.bst 
fetch needed a72e1ca8ea2ac7a5584f260983f9e940a75c842ea25399fd545dbbd60e0d9306 autotools/amhello.bst 
===============================================================================
[--:--:--][a72e1ca8][    pull:autotools/amhello.bst         ] START   test/autotools-amhello/a72e1ca8-pull.3472.log
[--:--:--][a72e1ca8][    pull:autotools/amhello.bst         ] STATUS  Pulling artifact a72e1ca8 <- http://localhost:7981
[--:--:--][a1e46db9][    pull:base/base-alpine.bst          ] START   test/base-base-alpine/a1e46db9-pull.3472.log
[--:--:--][378a7f2d][    pull:base.bst                      ] START   test/base/378a7f2d-pull.3472.log
[--:--:--][a1e46db9][    pull:base/base-alpine.bst          ] STATUS  Pulling artifact a1e46db9 <- http://localhost:7981
[--:--:--][378a7f2d][    pull:base.bst                      ] STATUS  Pulling artifact 378a7f2d <- http://localhost:7981
[--:--:--][a1e46db9][    pull:base/base-alpine.bst          ] STATUS  Pulling data for artifact a1e46db9 <- http://localhost:7982
[--:--:--][a72e1ca8][    pull:autotools/amhello.bst         ] STATUS  Pulling data for artifact a72e1ca8 <- http://localhost:7982
[--:--:--][378a7f2d][    pull:base.bst                      ] STATUS  Pulling data for artifact 378a7f2d <- http://localhost:7982
[--:--:--][378a7f2d][    pull:base.bst                      ] INFO    Pulled artifact 378a7f2d <- http://localhost:7982
[00:00:00][378a7f2d][    pull:base.bst                      ] SUCCESS test/base/378a7f2d-pull.3472.log
[--:--:--][378a7f2d][    push:base.bst                      ] START   test/base/378a7f2d-push.3472.log
[--:--:--][378a7f2d][    push:base.bst                      ] STATUS  Pushing data from artifact 378a7f2d -> http://localhost:7982
[--:--:--][378a7f2d][    push:base.bst                      ] INFO    Pushed data from artifact 378a7f2d -> http://localhost:7982
[--:--:--][378a7f2d][    push:base.bst                      ] STATUS  Pushing artifact 378a7f2d -> http://localhost:7981
[--:--:--][a72e1ca8][    pull:autotools/amhello.bst         ] INFO    Pulled artifact a72e1ca8 <- http://localhost:7982
[--:--:--][378a7f2d][    push:base.bst                      ] INFO    Remote (http://localhost:7981) already has artifact 378a7f2d cached
[00:00:00][378a7f2d][    push:base.bst                      ] SKIPPED Push
[00:00:00][a72e1ca8][    pull:autotools/amhello.bst         ] SUCCESS test/autotools-amhello/a72e1ca8-pull.3472.log
[--:--:--][a72e1ca8][    push:autotools/amhello.bst         ] START   test/autotools-amhello/a72e1ca8-push.3472.log
[--:--:--][a72e1ca8][    push:autotools/amhello.bst         ] STATUS  Pushing data from artifact a72e1ca8 -> http://localhost:7982
[--:--:--][a72e1ca8][    push:autotools/amhello.bst         ] INFO    Pushed data from artifact a72e1ca8 -> http://localhost:7982
[--:--:--][a72e1ca8][    push:autotools/amhello.bst         ] STATUS  Pushing artifact a72e1ca8 -> http://localhost:7981
[--:--:--][a72e1ca8][    push:autotools/amhello.bst         ] INFO    Remote (http://localhost:7981) already has artifact a72e1ca8 cached
[00:00:00][a72e1ca8][    push:autotools/amhello.bst         ] SKIPPED Push
[--:--:--][a1e46db9][    pull:base/base-alpine.bst          ] INFO    Pulled artifact a1e46db9 <- http://localhost:7982
[00:00:02][a1e46db9][    pull:base/base-alpine.bst          ] SUCCESS test/base-base-alpine/a1e46db9-pull.3472.log
[--:--:--][a1e46db9][    push:base/base-alpine.bst          ] START   test/base-base-alpine/a1e46db9-push.3472.log
[--:--:--][a1e46db9][    push:base/base-alpine.bst          ] STATUS  Pushing data from artifact a1e46db9 -> http://localhost:7982
[--:--:--][a1e46db9][    push:base/base-alpine.bst          ] INFO    Pushed data from artifact a1e46db9 -> http://localhost:7982
[--:--:--][a1e46db9][    push:base/base-alpine.bst          ] STATUS  Pushing artifact a1e46db9 -> http://localhost:7981
[--:--:--][a1e46db9][    push:base/base-alpine.bst          ] INFO    Remote (http://localhost:7981) already has artifact a1e46db9 cached
[00:00:00][a1e46db9][    push:base/base-alpine.bst          ] SKIPPED Push
[00:00:02][        ][    main:core activity                 ] SUCCESS Build

Pipeline Summary
    Total:       3
    Session:     3
    Pull Queue:  processed 3, skipped 0, failed 0 
    Fetch Queue: processed 0, skipped 3, failed 0 
    Build Queue: processed 0, skipped 3, failed 0 
    Push Queue:  processed 0, skipped 3, failed 0 
```

--- a/.github/run-ci.sh
+++ b/.github/run-ci.sh
@@ -63,6 +63,7 @@ function runTest() {
         --env-file ${topdir}/common.env \
         --file ${topdir}/compose/ci.docker-compose.yml \
         run "${test_name}"
+    return $?
 }
 
 
@@ -80,9 +81,12 @@ function runServiceTest() {
         --env-file "${topdir}/common.env" \
         --file "${topdir}/compose/ci.${test_name}.yml" \
         up --detach --renew-anon-volumes --remove-orphans
+
     docker-compose \
         --env-file "${topdir}/common.env" \
         --file "${topdir}/compose/ci.docker-compose.yml" run ${test_name}
+    test_exit_status=$?
+
     docker-compose \
         --env-file "${topdir}/common.env" \
         --file "${topdir}/compose/ci.${test_name}.yml" stop
@@ -91,31 +95,39 @@ function runServiceTest() {
         --file "${topdir}/compose/ci.${test_name}.yml" logs
     docker-compose \
         --env-file "${topdir}/common.env" \
-        --file "${topdir}/compose/ci.${test_name}.yml" down
+        --file "${topdir}/compose/ci.${test_name}.yml" down --volumes
+
+    return $test_exit_status
 }
 
 
-# Lazily ensure that the script exits when a command fails
-#
-set -e
-
 if [ -z "${test_names}" ]; then
-    runTest "lint"
-    runTest "mypy"
-    runTest "debian-10"
-    runTest "fedora-34"
-    runTest "fedora-35"
-    runTest "fedora-missing-deps"
-    runServiceTest "buildbarn"
-    runServiceTest "buildgrid"
+    for test_name in "mypy debian-10 fedora-34 fedora-35 fedora-missing-deps"; do
+	if ! runTest "${test_name}"; then
+	    echo "Tests failed"
+	    exit 1
+	fi
+    done
+    for test_name in "buildgrid buildbarn"; do
+	if ! runServiceTest "${test_name}"; then
+	    echo "Tests failed"
+	    exit 1
+	fi
+    done
 else
     if $arg_service; then
 	for test_name in "${test_names}"; do
-	    runServiceTest "${test_name}"
+	    if ! runServiceTest "${test_name}"; then
+		echo "Tests failed"
+		exit 1
+	    fi
 	done
     else
 	for test_name in "${test_names}"; do
-	    runTest "${test_name}"
+	    if ! runTest "${test_name}"; then
+		echo "Tests failed"
+		exit 1
+	    fi
 	done
     fi
 fi

